boa/explorer.py

```diff
@@ -1,14 +1,29 @@
-import json
 import time
 from typing import Optional
 
-import requests
+from boa.rpc import json
 
-SESSION = requests.Session()
+try:
+    from requests_cache import CachedSession, Response
+
+    def _filter_fn(res: Response) -> bool:
+        data = res.json()
+        return _is_success_response(data) and not _is_rate_limited(data)
```

Comment:
i think the `not _is_rate_limited(data)` clause is redundant, it will always return False if `_is_success_response(data)` returns true